flycheck-haskell.el

```diff
@@ -307,9 +318,10 @@ buffer."
                                   .autogen-directories)
                          (when (car .should-include-version-header)
                            '("-optP-include" "-optPcabal_macros.h"))
-                         (cons "-hide-all-packages"
+                         (when (not (car .package-env-exists))
```

Comment:
I'm actually slightly concerne about what would happen if we're using `ghc-stack` checker but package env file does exist (e.g. a project was recently built both with both tools)? Can you please add a test case for that?

Reply:
When the cabal configuration script is run with stuck, ghc still reads the package environment. I added a test to confirm it, but was failing, because the dependencies need to be passed explicitly when script is run through stack.